dff/context_storages/database.py

```diff
@@ -3,84 +3,175 @@
 --------
 Base module. Provided classes:
     - Abstract context storage interface :py:class:`.DBAbstractContextStorage`.
-    - An intermediate class to inherit from: :py:class:`.DBContextStorage`
 """
+import asyncio
 import importlib
 import threading
 from functools import wraps
 from abc import ABC, abstractmethod
-from typing import Any, Callable
+from typing import Callable, Hashable, Optional
 
 from .protocol import PROTOCOLS
+from ..script import Context
 
 
 class DBAbstractContextStorage(ABC):
```

Comment:
My opinion is better to hold `DBContextStorage` name instead `DBAbstractContextStorage`